ThirdParty/json.swift

```diff
@@ -58,33 +58,33 @@ extension JSON {
     }
     /// constructs JSON object from string
     public convenience init(string:String) {
-        let enc:NSStringEncoding = NSUTF8StringEncoding
-        self.init(data: string.dataUsingEncoding(enc)!)
+        let enc:String.Encoding = String.Encoding.utf8
+        self.init(data: string.data(using: enc)!)
     }
     /// parses string to the JSON object
     /// same as JSON(string:String)
-    public class func parse(string:String)->JSON {
+    public class func parse(_ string:String)->JSON {
         return JSON(string:string)
     }
     /// constructs JSON object from the content of NSURL
-    public convenience init(nsurl:NSURL) {
-        var enc:NSStringEncoding = NSUTF8StringEncoding
+    public convenience init(nsurl:URL) {
+        var enc:String.Encoding = String.Encoding.utf8
         do {
             let str = try String(NSString(
```

Comment:
no need to create `NSString` here. Can use `String(contentsOf: nsurl, encoding: enc)` instead